src/figanos/matplotlib/plot.py

```diff
@@ -2768,3 +2769,234 @@ def partition(
+        plot_kw = [plot_kw for _ in range(len(d))]
+
+    labels_x = da[other_dims[0]].values
+    labels_y = da[other_dims[1]].values
+    m, n = d[0].shape[0], d[0].shape[1]
+
+    # plot
+    if len(d) == 2:
+
+        x = np.arange(m + 1)
+        y = np.arange(n + 1)
+        xss, ys = np.meshgrid(x, y)
+        zs = (xss * ys) % 10
+        triangles1 = [
+            (i + j * (m + 1), i + 1 + j * (m + 1), i + (j + 1) * (m + 1))
+            for j in range(n)
+            for i in range(m)
+        ]
+        triangles2 = [
+            (i + 1 + j * (m + 1), i + 1 + (j + 1) * (m + 1), i + (j + 1) * (m + 1))
+            for j in range(n)
+            for i in range(m)
+        ]
+        triang1 = Triangulation(xss.ravel(), ys.ravel(), triangles1)
+        triang2 = Triangulation(xss.ravel(), ys.ravel(), triangles2)
+        triangul = [triang1, triang2]
+
+        imgs = [
+            ax.tripcolor(t, np.ravel(val), **plotkw)
+            for t, val, plotkw in zip(triangul, d, plot_kw)
+        ]
+
+        plt.xticks(np.array(range(m)) + 0.5, labels=labels_x, rotation=45)
+        plt.yticks(np.array(range(n)) + 0.5, labels=labels_y, rotation=90)
+
+    elif len(d) == 4:
+
+        xv, yv = np.meshgrid(
+            np.arange(-0.5, m), np.arange(-0.5, n)
+        )  # vertices of the little squares
+        xc, yc = np.meshgrid(
+            np.arange(0, m), np.arange(0, n)
+        )  # centers of the little squares
+        x = np.concatenate([xv.ravel(), xc.ravel()])
+        y = np.concatenate([yv.ravel(), yc.ravel()])
+        cstart = (m + 1) * (n + 1)  # indices of the centers
+
+        triangles_n = [
+            (i + j * (m + 1), i + 1 + j * (m + 1), cstart + i + j * m)
+            for j in range(n)
+            for i in range(m)
+        ]
+        triangles_e = [
+            (i + 1 + j * (m + 1), i + 1 + (j + 1) * (m + 1), cstart + i + j * m)
+            for j in range(n)
+            for i in range(m)
+        ]
+        triangles_s = [
+            (i + 1 + (j + 1) * (m + 1), i + (j + 1) * (m + 1), cstart + i + j * m)
+            for j in range(n)
+            for i in range(m)
+        ]
+        triangles_w = [
+            (i + (j + 1) * (m + 1), i + j * (m + 1), cstart + i + j * m)
+            for j in range(n)
+            for i in range(m)
+        ]
+        triangul = [
+            Triangulation(x, y, triangles)
+            for triangles in [triangles_n, triangles_e, triangles_s, triangles_w]
+        ]
+
+        imgs = [
+            ax.tripcolor(t, np.ravel(val), **plotkw)
+            for t, val, plotkw in zip(triangul, d, plot_kw)
+        ]
+        plt.xticks(np.array(range(m)), labels=labels_x, rotation=45)
+        plt.yticks(np.array(range(n)), labels=labels_y, rotation=90)
+
+    else:
+        raise ValueError(
+            f"The length of the dimensiondim ({dim},{len(d)}) should be either 2 or 4. It represents the number of triangles."
+        )
+
+    plt.xlabel(other_dims[0])
```

Comment:
Could xlabel or ylabel also be linked to use_attrs?

Reply:
yes!